### PR TITLE
Firebolt connector docs

### PR DESCRIPTION
**Pull request by oliviamiannone**
**Description:**

Adds docs for new Firebolt connector. 

**Notes for reviewers:**

Following standard process, and no big outstanding questions on my end. As usual, needs fact-checking, and let me know if I've missed anything important.

---
This change is [Reviewable](https://reviewable.io/reviews/estuary/flow/426)

### CODE REVIEW

File: site/docs/reference/Connectors/materialization-connectors/Firebolt.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+#### Endpoint
+
+| Property | Title | Description | Type | Required/Default |
+|---|---|---|---|---|
+| `/aws_key_id` | AWS Key ID | AWS Key ID for accessing the S3 bucket. | string |  |
+| `/aws_region` | AWS Region | AWS Region the bucket is in. | string |  |
+| `/aws_secret_key` | AWS Secret Key | AWS Secret Key for accessing the S3 bucket. | string |  |
+| **`/database`** | Database | Name of the Firebolt database. | string | Required |
+| **`/engine_url`** | Engine URL | Engine URL of the Firebolt database, in the format: `<engine-name>.<organization>.<region>.app.firebolt.io`. | string | Required |
```
Comment (mdibaiee) [inline site/docs/reference/Connectors/materialization-connectors/Firebolt.md:39]: Did the title and description end up being auto-generated for Firebolt or were they still missing?

Comment (oliviamiannone) [reply site/docs/reference/Connectors/materialization-connectors/Firebolt.md:39]: They were auto-generated! Just tweaked a couple small things for formatting

File: site/docs/reference/Connectors/materialization-connectors/Firebolt.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# Firebolt
+
+This Flow connector materializes [delta updates](../../../concepts/materialization.md#delta-updates) of Flow collections into Firebolt `FACT` or `DIMENSION` tables.
+
+To interface between Flow and Firebolt, the connector creates an _external table_:
+a virtual table in an S3 bucket used as a staging area.
```
Comment (mdibaiee) [inline site/docs/reference/Connectors/materialization-connectors/Firebolt.md:6]: I would reword this to be
```suggestion
a table connected to an S3 bucket, acting as a SQL interface to JSON documents stored in the S3 bucket. (See [Working with external tables](https://docs.firebolt.io/loading-data/working-with-external-tables.html))
```

Since the table itself is in Firebolt, but it acts as a SQL interface on over the JSON documents there.

File: site/docs/reference/Connectors/materialization-connectors/Firebolt.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+[`ghcr.io/estuary/materialize-firebolt:dev`](https://ghcr.io/estuary/materialize-firebolt:dev) provides the latest connector image. You can also follow the link in your browser to see past image versions.
+
+## Prerequisites
+
+To use this connector, you'll need:
+
+* A Firebolt database with at least one [engine](https://docs.firebolt.io/working-with-engines/working-with-engines.html)
+* An S3 bucket in which to stage the external table
```
Comment (mdibaiee) [inline site/docs/reference/Connectors/materialization-connectors/Firebolt.md:15]:
```suggestion
* An S3 bucket where the JSON documents will be stored before being inserted to the main Firebolt table (see Firebolt's [Loading data](https://docs.firebolt.io/loading-data/loading-data.html))
```

File: site/docs/reference/Connectors/materialization-connectors/Firebolt.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            s3_bucket: my-bucket
+            username: firebolt-user
+          # Path to the latest version of the connector, provided as a Docker image
+          image: ghcr.io/estuary/materialize-firebolt:dev
+	# If you have multiple collections you need to materialize, add a binding for each one
+    # to ensure complete data flow-through
+    bindings:
+      - resource:
+          table: table-name
+          table_type: FACT
```
Comment (mdibaiee) [inline site/docs/reference/Connectors/materialization-connectors/Firebolt.md:74]: I actually think the code I wrote currently is case-sensitive lowercase and might not work with uppercase, might be worth changing that to be case-insensitive but just in case let's keep it lowercase for now:

```suggestion
          table_type: fact
```

**Review by mdibaiee — APPROVED** [review]
Thanks a million for the quick turnover Olivia! A few suggestions below

File: site/docs/reference/Connectors/materialization-connectors/Firebolt.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
+# Firebolt
+
+This Flow connector materializes [delta updates](../../../concepts/materialization.md#delta-updates) of Flow collections into Firebolt `FACT` or `DIMENSION` tables.
+
+To interface between Flow and Firebolt, the connector creates an _external table_:
```
Comment (mdibaiee) [inline site/docs/reference/Connectors/materialization-connectors/Firebolt.md:5]:
```suggestion
To interface between Flow and Firebolt, the connector creates a Firebolt _external table_:
```

File: site/docs/reference/Connectors/materialization-connectors/Firebolt.md (hunk 4 above)
Comment (oliviamiannone) [reply site/docs/reference/Connectors/materialization-connectors/Firebolt.md:74]: Got it, thanks, making that change. I assumed uppercase because Firebolt's docs refer to it in uppercase consistently, so I agreed it could be helpful to make it case-insensitive at some point.

**Comment by oliviamiannone** [comment]
@mdibaiee Thanks for helping me clarify those firebolt concepts. Could you take one more look at the most recent commit? instead of taking the suggestions verbatim I thought it'd be best to consolidate all the explanations of Firebolt's loading in one place, so I want to make sure what I ended up with is still right.

**Review by mdibaiee — APPROVED** [review]
Thanks Olivia, it's much clearer now!
